### ISSUE TITLE
Traits 6.1.0 deprecation warning

### ISSUE THREAD

**Issue by ericpre** [enthought/traitsui#883]
For an example of deprecation warning, see https://travis-ci.org/github/ericpre/hyperspy/jobs/696413265

```
  /home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/traitsui/view.py:111: DeprecationWarning: 'TraitPrefixList' trait handler has been deprecated. Use PrefixList instead.
    TraitPrefixList(
```
and 
```
  /home/travis/virtualenv/python3.8.0/lib/python3.8/site-packages/traits/traits.py:330: DeprecationWarning: 'TraitMap' trait handler has been deprecated. Use Map instead.
    other.append(TraitMap(map))
```

**Comment by mdickinson** [comment]
Thank you for the report! This is on our roadmap to fix for TraitsUI 7.1, which is currently scheduled for early next quarter. We unfortunately can't fix this in a TraitsUI bugfix release, since the fix *requires* Traits >= 6.1.0, and bumping the dependent Traits version in a TraitsUI bugfix release wouldn't be a friendly thing to do.

We did go back-and-forth about whether to delay these deprecation warnings until Traits 6.2.0, for exactly this reason: it's annoying to get deprecation warnings about something you don't control. If the warnings are sufficiently troublesome, that may still be an option - we could make a Traits 6.1.1 bugfix release with the warnings silenced. Otherwise, I'd suggest using `filterwarnings` to filter these warnings out until TraitsUI 7.1 is released.

**Comment by rahulporuri** [comment]
closing as the deprecation warning was addressed in #1053 and the compat layer will be removed in #1599 . we already opened #1296 to handle the removal of the compat layer.
